src/widgets/ProfileButton/ProfileButton.tsx

```diff
@@ -1,10 +1,20 @@
-import {Button} from "@/shared";
-import {useMyDetailsQuery} from "@/entities/profile/api/profileApi.ts";
-
+// import { Button } from "@/shared";
+import { useMyDetailsQuery } from "@/entities/profile/api/profileApi.ts";
+import { Menu } from "@headlessui/react";
+import { BurgerMenu } from "@/features/burgerMenu";
 
 export function ProfileButton() {
-    let {data: profileData} = useMyDetailsQuery()
-    return (
-        <Button HTMLType='button' type='secondary' extraClass='text-base mt-2.5'>{profileData?.username}</Button>
-    )
+  let { data: profileData } = useMyDetailsQuery();
+  return (
+    <Menu>
+      {({ open }) => (
```

Comment:
i think you can delete this open state because headless-ui state no need to control manually, it automatically work

Reply:
fixed